### DISCUSSION TITLE
nhl analytics resources

### DISCUSSION THREAD

**Discussion by dmf95** [the-data-base/nhl-data#19] [General]
below are some of the better resources I've come across as it relates to hockey analytics. It is not all-encompassing and is also broad in the sense that some resources offer more than a single kind of hockey resource... 

* [NHL Stats API gitlab](https://gitlab.com/dword4/nhlapi/-/blob/master/stats-api.md)
* [NHL](https://www.nhl.com/stats/teams) 
* [Natural Stat Trick](http://www.naturalstattrick.com/)
* [Money Puck](https://moneypuck.com/data.htm)
* [Evolving Hockey](https://evolving-hockey.com/) - note: @dmf95 pays for a monthly subscription, msg him for login creds
* [Corey Sznajder's Tableau](https://public.tableau.com/app/profile/corey.sznajder)
* [Puck Pedia](https://puckpedia.com/stats)
* [Puck on net](http://www.puckon.net/)
* [Hockey viz](http://hockeyviz.com/)
* [Shift chart](http://www.shiftchart.com/)
* [Hockey reference](https://www.hockey-reference.com/)
* [Cap friendly](https://www.capfriendly.com/)
* [Play by play project Simon Columbe](https://github.com/SimonCoulombe/nhl_play_by_play)

here are some articles that help explain hockey analytics metrics
* [Article on advanced hockey metrics](https://thehockeywriters.com/hockey-advanced-analytics-importance-metrics-definitions/)
* [Sportsnet, explaining hockey analytics terms](https://www.sportsnet.ca/hockey/nhl/corsi-pdo-explaining-hockeys-analytics-terms/)
* [The Athletics, advanced stats primer](https://theathletic.com/121980/2017/10/09/an-advanced-stat-primer-understanding-basic-hockey-metrics/)
* [NHL, understanding advanced stats](https://www.nhl.com/lightning/news/hockey-analytics-101-understanding-advanced-stats-and-how-theyre-measured/c-735819)
* [Beginner's guide to advanced hockey stats](https://sites.northwestern.edu/nusportsanalytics/2020/05/06/advanced-hockey-statistics/)

NOTE-- it is important to distinguish that this is public hockey analytics and that teams in the NHL have access to better and more comprehensive data

**Comment by gavh3** [comment]
Let's identify which of these have a callable API, or if its an analytical layer on top of the API then what the source is

**Reply by dmf95** [reply]
good idea. Linking the [Google Sheet](https://docs.google.com/spreadsheets/d/1Xek1I7qjAeqXljoxsglGm9VueVSCOoGu5QkRlX15wbY/edit#gid=1410390943) that you started for this

**Comment by dmf95** [comment]
EUREKA! I believe I have found the shift-level data. Thanks to @nkumar97 who led me down the rabbit hole: 
* Test 1:  https://api.nhle.com/stats/rest/en/shiftcharts?cayenneExp=gameId=2020030415
* Test 2:  https://api.nhle.com/stats/rest/en/shiftcharts?cayenneExp=gameId=2020030316
* Test 3:  https://api.nhle.com/stats/rest/en/shiftcharts?cayenneExp=gameId=2020030312

@gavh3 - I believe the link @nkumar97 shared provides some guy's R logic for handling the data for inspo... can you confirm that this is doable for our singer tap?


FYI - Here is the rabbit hole I went down, some handy links along the way so sharing for visibility...
Started here: [Using the NHL API – Play by Play and Shift Data](http://www.hainke.ca/index.php/2019/11/20/using-the-nhl-api-play-by-play-and-shift-data/), thanks to @nkumar97 . The author's code didn't run on the few test example, but I somehow fiddled my way to [this Github repo](https://github.com/SimonCoulombe/nhl_play_by_play), which sent me this [random a$$ reddit thread](https://www.reddit.com/r/hockey/comments/bvn933/issue_with_nhl_shift_data_for_a_game_between_the/). Turns out, that reddit thread had a commenter with what we needed!
